### ISSUE TITLE
Error fetching OMEMO key

### ISSUE THREAD

**Issue by sunnat629** [siacs/Conversations#2672]
#### General information

* **Version:** 1.21.0
* **Device:** Google Pixel and Galaxy S6
* **Android Version:** Android 8.0.0 Stock and Android 5.1.1 Stock
* **Server name:** self-hosted
* **Server software:** Openfire 4.0.0
* **Installed server modules:** None
* **Conversations source:** self-build

#### Steps to reproduce

1. Install Application
2. Connect to Self Hosted Server
3. Registration on both devices with different userID
4. Add the other user
5. In the chatbox, select OMEMO option from both sides.

#### Expected result

What is the expected output?
-- We will able to chat with OMEMO Protocol

#### What do you see instead?

Can't Communicate with this protocol. Stuck in OMEMO Fingerprints activity
There are several errors. 
       * There are no usable keys available for this contact.\nFetching new keys from the server has been unsuccessful. Maybe there is something wrong with your contact’s server.
       * There are no usable keys available for this contact.\nMake sure you have mutual presence subscription.
       * There are no usable keys available for this contact. If you have purged any of their keys, they need to generate new ones.

#### Debug output

**adb Logcat from S6**
10-31 17:11:58.681 20183-20183/eu.siacs.conversations D/ViewRootImpl: ViewPostImeInputStage ACTION_DOWN
10-31 17:11:58.721 20183-20183/eu.siacs.conversations D/conversations: AxolotlService (1000@192.168.30.247): Finding devices without session for 1001@192.168.30.247
10-31 17:11:58.721 20183-20183/eu.siacs.conversations D/conversations: 1000@192.168.30.247: createSessionsIfNeeded() - jids with empty device list: [1001@192.168.30.247]
10-31 17:11:58.721 20183-20183/eu.siacs.conversations D/conversations: 1000@192.168.30.247: fetching device ids for 1001@192.168.30.247
10-31 17:11:58.721 20183-20183/eu.siacs.conversations I/Timeline: Timeline: Activity_launch_request id:eu.siacs.conversations time:21547799
10-31 17:11:58.731 3694-4975/? V/WindowManager: addAppToken: AppWindowToken{3d4fa39e token=Token{29433dd9 ActivityRecord{2d964020 u0 eu.siacs.conversations/.ui.TrustKeysActivity t423}}} to stack=1 task=423 at 1
10-31 17:11:58.731 3694-4975/? I/ActivityManager: Skip updateThumbnail for r=ActivityRecord{3c509b93 u0 eu.siacs.conversations/.ui.ConversationActivity t423}
10-31 17:11:58.731 3694-4975/? V/ApplicationPolicy: isApplicationStateBlocked userId 0 pkgname eu.siacs.conversations
10-31 17:11:58.731 3694-4975/? V/ApplicationPolicy: isApplicationStateBlocked userId 0 pkgname eu.siacs.conversations
10-31 17:11:58.741 20183-21236/eu.siacs.conversations I/conversations: AxolotlService (1000@192.168.30.247): Creating axolotl sessions if needed...
10-31 17:11:58.741 20183-21236/eu.siacs.conversations D/conversations: AxolotlService (1000@192.168.30.247): Finding devices without session for 1001@192.168.30.247
10-31 17:11:58.751 20183-20183/eu.siacs.conversations D/PhoneWindow: *FMB* installDecor mIsFloating : false
10-31 17:11:58.751 20183-20183/eu.siacs.conversations D/PhoneWindow: *FMB* installDecor flags : -2139029248
10-31 17:11:58.761 20183-20183/eu.siacs.conversations D/Activity: performCreate Call Injection manager
10-31 17:11:58.761 3694-4939/? D/ActivityManager: startService callerProcessName:eu.siacs.conversations, calleePkgName: eu.siacs.conversations
10-31 17:11:58.761 3694-4939/? D/ActivityManager: retrieveServiceLocked(): component = eu.siacs.conversations/eu.siacs.conversations.services.XmppConnectionService; callingUser = 0; userId(target) = 0
10-31 17:11:58.761 3694-3730/? D/ActivityManager: bindService callerProcessName:eu.siacs.conversations, calleePkgName: eu.siacs.conversations, action: ui
10-31 17:11:58.761 3694-3730/? D/ActivityManager: retrieveServiceLocked(): component = eu.siacs.conversations/eu.siacs.conversations.services.XmppConnectionService; callingUser = 0; userId(target) = 0
10-31 17:11:58.761 20183-20183/eu.siacs.conversations I/InjectionManager: dispatchOnViewCreated > Target : eu.siacs.conversations.ui.TrustKeysActivity isFragment :false
10-31 17:11:58.761 20183-20183/eu.siacs.conversations D/SecWifiDisplayUtil: Metadata value : SecSettings2
10-31 17:11:58.761 3694-4526/? V/WindowManager: Adding window Window{248e5baa u0 d0 eu.siacs.conversations/eu.siacs.conversations.ui.TrustKeysActivity} at 14 of 21 (after Window{2cb96385 u0 d0 eu.siacs.conversations/eu.siacs.conversations.ui.ConversationActivity})
10-31 17:11:58.761 20183-20183/eu.siacs.conversations D/PhoneWindow: *FMB* isFloatingMenuEnabled mFloatingMenuBtn : null
10-31 17:11:58.761 20183-20183/eu.siacs.conversations D/PhoneWindow: *FMB* isFloatingMenuEnabled return false
10-31 17:11:58.771 3694-3773/? D/StatusBarManagerService: manageDisableList userId=0 what=0x0 pkg=Window{248e5baa u0 d0 eu.siacs.conversations/eu.siacs.conversations.ui.TrustKeysActivity}
10-31 17:11:58.791 20183-20183/eu.siacs.conversations D/SRIB_DCS: log_dcs ThreadedRenderer::initialize entered! 
10-31 17:11:58.791 20183-20474/eu.siacs.conversations D/mali_winsys: new_window_surface returns 0x3000,  [1440x2560]-format:1
10-31 17:11:58.801 20183-20183/eu.siacs.conversations I/InjectionManager: dispatchCreateOptionsMenu :eu.siacs.conversations.ui.TrustKeysActivity
10-31 17:11:58.801 20183-20183/eu.siacs.conversations I/InjectionManager: dispatchPrepareOptionsMenu :eu.siacs.conversations.ui.TrustKeysActivity
10-31 17:11:58.801 20183-20183/eu.siacs.conversations I/InjectionManager: dispatchPrepareOptionsMenu :eu.siacs.conversations.ui.TrustKeysActivity
10-31 17:11:58.801 20183-20183/eu.siacs.conversations I/InjectionManager: dispatchCreateOptionsMenu :eu.siacs.conversations.ui.ConversationActivity
10-31 17:11:58.801 20183-20183/eu.siacs.conversations I/InjectionManager: dispatchPrepareOptionsMenu :eu.siacs.conversations.ui.ConversationActivity
10-31 17:11:58.821 3694-4527/? I/MultiPhoneWindowManager: Changed -> mCurrentInputMethodTarget : Window{248e5baa u0 d0 eu.siacs.conversations/eu.siacs.conversations.ui.TrustKeysActivity}
10-31 17:11:58.841 20183-20183/eu.siacs.conversations I/Timeline: Timeline: Activity_idle id: android.os.BinderProxy@1632e779 time:21547919
10-31 17:11:59.091 3694-3775/? I/Timeline: Timeline: Activity_windows_visible id: ActivityRecord{2d964020 u0 eu.siacs.conversations/.ui.TrustKeysActivity t423} time:21548165
10-31 17:11:59.101 20183-20183/eu.siacs.conversations V/ActivityThread: updateVisibility : ActivityRecord{2a34e4b token=android.os.BinderProxy@669e8b {eu.siacs.conversations/eu.siacs.conversations.ui.ConversationActivity}} show : false

**adb Logcat from Pixel**
10-31 17:44:14.971 6055-6055/eu.siacs.conversations D/conversations: AxolotlService (1001@192.168.30.247): Finding devices without session for 1000@192.168.30.247
10-31 17:44:14.973 6055-6055/eu.siacs.conversations D/conversations: AxolotlService (1001@192.168.30.247): Found device 1000@192.168.30.247:921912424
10-31 17:44:14.973 6055-6055/eu.siacs.conversations D/conversations: AxolotlService (1001@192.168.30.247): skipping over 1000@192.168.30.247:921912424 because it's broken
10-31 17:44:14.975 6055-6055/eu.siacs.conversations D/conversations: 1001@192.168.30.247: createSessionsIfNeeded() - jids with empty device list: []
10-31 17:44:14.975 6055-6055/eu.siacs.conversations I/conversations: AxolotlService (1001@192.168.30.247): Creating axolotl sessions if needed...
10-31 17:44:14.975 6055-6055/eu.siacs.conversations D/conversations: AxolotlService (1001@192.168.30.247): Finding devices without session for 1000@192.168.30.247
10-31 17:44:14.977 6055-6055/eu.siacs.conversations D/conversations: AxolotlService (1001@192.168.30.247): Found device 1000@192.168.30.247:921912424
10-31 17:44:14.977 6055-6055/eu.siacs.conversations D/conversations: AxolotlService (1001@192.168.30.247): skipping over 1000@192.168.30.247:921912424 because it's broken
                                                                     
                                                                     --------- beginning of system
10-31 17:44:14.979 948-1407/? I/ActivityManager: START u0 {cmp=eu.siacs.conversations/.ui.TrustKeysActivity (has extras)} from uid 10504
10-31 17:44:15.084 948-993/? I/ActivityManager: Displayed eu.siacs.conversations/.ui.TrustKeysActivity: +92ms

**openfire log**
2017.10.31 07:05:04 org.jivesoftware.openfire.nio.ConnectionHandler - Closing connection due to exception in session: (0x0000025F: nio socket, server, /192.168.30.118:46372 => /192.168.30.247:5222)
javax.net.ssl.SSLHandshakeException: SSL handshake failed.
at org.apache.mina.filter.ssl.SslFilter.messageReceived(SslFilter.java:487)
at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
at org.apache.mina.core.filterchain.DefaultIoFilterChain.access$1200(DefaultIoFilterChain.java:47)
at org.apache.mina.core.filterchain.DefaultIoFilterChain$EntryImpl$1.messageReceived(DefaultIoFilterChain.java:765)
at org.apache.mina.core.filterchain.IoFilterAdapter.messageReceived(IoFilterAdapter.java:109)
at org.apache.mina.core.filterchain.DefaultIoFilterChain.callNextMessageReceived(DefaultIoFilterChain.java:417)
at org.apache.mina.core.filterchain.DefaultIoFilterChain.fireMessageReceived(DefaultIoFilterChain.java:410)
at org.apache.mina.core.polling.AbstractPollingIoProcessor.read(AbstractPollingIoProcessor.java:710)
at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:664)
at org.apache.mina.core.polling.AbstractPollingIoProcessor.process(AbstractPollingIoProcessor.java:653)
at org.apache.mina.core.polling.AbstractPollingIoProcessor.access$600(AbstractPollingIoProcessor.java:67)
at org.apache.mina.core.polling.AbstractPollingIoProcessor$Processor.run(AbstractPollingIoProcessor.java:1124)
at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
Caused by: javax.net.ssl.SSLException: Received fatal alert: certificate_unknown
at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1666)
at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1634)
at sun.security.ssl.SSLEngineImpl.recvAlert(SSLEngineImpl.java:1800)
at sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:1083)
at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:907)
at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:781)
at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624)
at org.apache.mina.filter.ssl.SslHandler.unwrap(SslHandler.java:728)
at org.apache.mina.filter.ssl.SslHandler.unwrapHandshake(SslHandler.java:666)
at org.apache.mina.filter.ssl.SslHandler.handshake(SslHandler.java:552)
at org.apache.mina.filter.ssl.SslHandler.messageReceived(SslHandler.java:351)
at org.apache.mina.filter.ssl.SslFilter.messageReceived(SslFilter.java:468)

**Comment by licaon-kter** [comment]
Use https://gist.github.com to paste the logs and post the link here instead of this wall of text.

Openfire 4.0.0 or Openfire 4.1.6 (aka latest stable) ?

**Comment by iNPUTmice** [comment]
If you are looking for free support please try the community group chat.

**Comment by sunnat629** [comment]
@licaon-kter 
--- I tried the latest version (Openfire 4.1.6), but only in Android 8.0.0, I am able to login. I couldn't connect to Android 5 or 6. I had to downgrade my server to openfire 4.0.0.
 
--- It works with OTR and Unencrypted both.
--- We have an internal server where I installed openfire 4.0.0, it has no external excess. Yes, I am hitting IPs directly. 
I know, this is not a place for this discussion, so can I have your mailid, or an appointment to discuss with you about this.
